src/View/View.php

```diff
@@ -1249,7 +1249,7 @@ protected function _getViewFileName($name = null)
         if (strlen($this->subDir)) {
             $subDir = $this->subDir . DIRECTORY_SEPARATOR;
             // Check if templatePath already terminates with subDir
-            if (strrpos($templatePath, $subDir) == strlen($templatePath) - strlen($subDir)) {
+            if ($templatePath != $subDir && substr($templatePath, -(strlen($subDir))) == $subDir) {
```

Comment:
You can use `===` when you are comparing strings now.

Reply:
Shouldn't  `$templatePath` and `$subDir`   be equal sometimes ?

Reply:
When would that happen? Extension directories are generally lower case, like `csv` while template paths usually end with the controller name `Csvs`.

Reply:
@dereuromark - yes, maybe `===` should be used.

@saeideng - I agree with mark that is shouldn't happen because `$templatePath` is using the controller name and the first letter is uppercase, whereas the `$subDir` is typically all lowercase.

@markstory - Wouldn't the test you added during this commit (https://github.com/cakephp/cakephp/commit/115a3547ab0c9120e3d2c6a3a2e6d2c3573ddc67) cover this case as well?

Reply:
It does to some extent. But you have a new scenario that requires different code. With only the existing test, this change could be reverted and there wouldn't be a failing test to let us know there was a regression.